encryption/src/main/java/org/apache/solr/encryption/EncryptionTransactionLog.java

```diff
@@ -156,7 +169,7 @@ private static int readBEInt(ReadableByteChannel channel, ByteBuffer readBuffer)
   }
 
   /** Supplies and releases {@link EncryptionDirectory}. */
-  protected interface EncryptionDirectorySupplier {
+  public interface EncryptionDirectorySupplier {
```

Comment:
Note that Solr uses `RefCounted` for this scenario in a number of cases.  This could be another?

Reply:
Not in this case, this interface is here to isolate the SolrCore needed to get the EncryptionDirectory.
EncryptionUpdateLog knows the SolrCore, but does not pass it to the EncryptionTransactionLog, instead it passes this interface. And the release method calls DirectoryFactory.release(Directory), required when done with the Directory.